### DISCUSSION TITLE
does wiki js support pasting an image from clipboard?

### DISCUSSION THREAD

**Discussion by genfoch01** [requarks/wiki#6714] [Help / Questions]
first attempt to install/use wiki js

just installed wiki js on docker. system shows I have 2.5.3 installed 

I copied an image to my clipboard but can't paste it into the page I am editing. 

if the image is copied from a a web site ( which I guess means it has a url associated with it ) it works. 

wiki js appears to be using ckeditor 5.  taking a look at [stack overflow](https://stackoverflow.com/questions/63591824/ck-editor-version-5-paste-screenshot-not-working-for-react-js) it appears the editor supports it but it sounds like you need to add several plugins to make this work. 

Is pasting from the clipboard supported?  if so can someone post a link on how I can enable this? 
thanks for your time, 
GF

**Comment by FarmDoc** [comment]
the very great draw.io Tool in the Markdown-Editor does support Pasting.

Also I read somewhere, that V3.0 of wiki.js (sometime this year) will support pasting also in the WYSIWYG Editor.

**Reply by genfoch01** [reply]
when I convert the page to markdown  I paste my image into the editor and the image flashes there but then disappears like it was never pasted there at all.   when I saved the page nothing is changed, I can view the page but there is no image. 

![image](https://github.com/requarks/wiki/assets/144620889/6017c00e-8871-4617-80b5-f43e24264819)

**Reply by FarmDoc** [reply]
It cannot be pasted directly into the markdown.

You have to click on the  third symbol on the right sideframe to open the draw.io-Tool ("1" in my image)
There you can paste and also add things like arrows, highlights, etc.

![image](https://github.com/requarks/wiki/assets/26462002/83cf09e7-8245-4f83-978d-048211d65979)

Of course you can also use the normal upload of images ("2" in my image), but I find that only useful for images that are needed at multiple places in the wiki

**Reply by genfoch01** [reply]
Thanks for the response! That worked thank you so much.  I wonder why this is not in the visual editor.

**Comment by genfoch01** [comment]
thanks FarmDoc.  I noticed when I added the image in markdown, when I converted the page to visual editor the image was removed from the page.  when I switched the page back to markdown the image was still gone.

**Reply by FarmDoc** [reply]
Yes, that is true. No converting.

As mentioned above, the visual Edit should get image pasting with 3.0 sometine this year.

**Comment by genfoch01** [comment]
I version 3's release date:   ETA: H2 2023 .  H2? second half of the year?

**Reply by FarmDoc** [reply]
Yes, this is how I understand it
